From 16d584f3387f0b4f93d69290143cd38fa3506a91 Mon Sep 17 00:00:00 2001
From: Nitish Bidwe <nitish.bidwe@lge.com>
Date: Fri, 8 Apr 2022 17:16:46 +0530
Subject: [PATCH] Fix connman crash on p2p/connect

:Release Notes:
Fix connman crash on p2p/connect

:Detailed Notes:
Connman crasing on p2p/connect due to no scope of
funcation

:Testing Performed:
Build successfully.
p2p connection successfully.

:QA Notes:
N/A

:Issues Addressed:
[WRN-16359] [webOS OSE]: Connected status is displayed
            false

Upstream-Status: Inappropriate [webOS specific]
---
 gsupplicant/gsupplicant.h | 1 +
 src/connman.h             | 2 ++
 2 files changed, 3 insertions(+)

diff --git a/gsupplicant/gsupplicant.h b/gsupplicant/gsupplicant.h
index c787d5a..671ea4e 100644
--- a/gsupplicant/gsupplicant.h
+++ b/gsupplicant/gsupplicant.h
@@ -661,6 +661,7 @@ GSupplicantInterface *g_supplicant_group_get_orig_interface(GSupplicantGroup *gr
 char *g_supplicant_group_get_object_path(GSupplicantGroup *group);
 char *g_supplicant_group_get_bssid_no_colon(GSupplicantGroup *group);
 GSupplicantGroup *g_supplicant_get_group(const char *path);
+GSupplicantP2PPersistentGroup* g_supplicant_interface_get_p2p_persistent_group(GSupplicantInterface *interface, GSupplicantGroup *group);
 
 int g_supplicant_group_get_role(GSupplicantGroup *group);
 char *g_supplicant_group_get_ssid(GSupplicantGroup *group);
diff --git a/src/connman.h b/src/connman.h
index 3badeb7..4160c2c 100644
--- a/src/connman.h
+++ b/src/connman.h
@@ -1131,3 +1131,5 @@ unsigned int __connman_util_random_delay_ms(unsigned int secs);
 int __connman_util_init(void);
 void __connman_util_cleanup(void);
 char *__connman_util_insert_colon_to_mac_addr(const char *mac_addr);
+char *__connman_util_mac_binary_to_string_no_colon(const unsigned char binmac[6]);
+const char * g_supplicant_peer_identifier_from_intf_address(const char* pintf_addr);
-- 
2.17.1

